src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -376,7 +425,7 @@ public void setAzureCloud(AzureVMCloud cloud) {
         azureCloud = cloud;
         if (StringUtils.isBlank(storageAccountName)) {
```

Comment:
Should check `storageAccountNameReferenceType` here? If it's existing and storageAccountName is empty, should throw error.